kbfscrypto/encrypted_data.go

```diff
@@ -164,7 +186,20 @@ func EncryptEncodedPrivateMetadata(encodedPrivateMetadata []byte, key TLFCryptKe
 func DecryptPrivateMetadata(
 	encryptedPrivateMetadata EncryptedPrivateMetadata, key TLFCryptKey) (
 	[]byte, error) {
-	return decryptData(encryptedPrivateMetadata.encryptedData, key.Data())
+	if encryptedPrivateMetadata.encryptedData.Version ==
+		EncryptionSecretboxWithKeyNonce {
+		// Only blocks should have v2 encryption.
+		return nil, errors.WithStack(InvalidEncryptionVer{
+			encryptedPrivateMetadata.encryptedData.Version})
+	}
+
+	nonce, err := encryptedPrivateMetadata.encryptedData.Nonce24()
```

Comment:
ah, what I was thinking was that, here and in the other cases where `Version != EncryptionSecretboxWithKeyNonce`, you'd just pass in `[24]byte{}` as the nonce. You'd save a copy, and if someone messes up and passes in a zero nonce when they shouldn't, they'll get an error.

Reply:
But I need to convert `encryptedData.Nonce` to a `[24]byte` anyway, to pass into `secretbox.Seal`.  This is just moving the copy to an earlier function, but I don't think it's wasted.

Reply:
Oh, maybe you're saying because the `[24]byte` is copied when passed between functions?  Is the `[24]byte{}` not copied?

Reply:
Ah, I see what you mean. That seems fine, then!